### PR TITLE
Fix to ensure a PersonPage exists and is published before we can link to them

### PR DESCRIPTION
**Pull request by engineervix**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1369626170)

### Description of Changes Made

This ticket adds a fix to the author display component used on `WorkPage`s, `HistoricalWorkPage`s and `BlogPage`s -- ensuring that the `author.person_page` not only exists but is also published.

Since the display of author information is dependent on the Author / PersonPage logic, I wrote some unit tests to help catch regressions in this logic in the future. 

### How to Test

1. Pick a blog page / work page / historical work page whose author links to the author's person page.
2. Edit that person page by unpublishing it
3. Go back to the page in 1, the author should still be displayed, but there shouldn't be any link to the author's person page, since it's no longer live.
4. Now, delete the person page
5. Repeat step 3

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [x] Added
- [ ] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on Ubuntu
  - Latest version of Firefox on Ubuntu

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

**Review by ianmeigh — COMMENTED** [review]
These changes look good to me, and the tests are a great addition! I'll add Nick as a reviewer for approval.

File: tbx/people/tests/test_author.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+class TestAuthor(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        site = Site.objects.get(is_default_site=True)
+        root = site.root_page.specific
+        home = HomePageFactory(parent=root)
+
+        site.root_page = home
```
Comment (nimasmi) [inline tbx/people/tests/test_author.py:18]: **Comment** (nitpick): It's a little odd, but not at all disallowed, to have a site pointing at a non-top level page. I'm sure it's inconsequential, but you can add a home page at the top level (well, one below the treebeard root page) with:

```suggestion
        site = Site.objects.get(is_default_site=True)
        root = Page.objects.get(depth=1)  # or .get(path="0001")
        home = HomePageFactory(parent=root)

        site.root_page = home
```

File: tbx/people/tests/test_author.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+from django.test import TestCase
+
+from tbx.core.factories import HomePageFactory
+from tbx.images.factories import CustomImageFactory
+from tbx.people.models import Author, PersonIndexPage, PersonPage
```
Comment (nimasmi) [inline tbx/people/tests/test_author.py:5]: **Suggestion**: Factories for these might make your life easier in future, too. See how much more verbose the PersonPage creation steps are in the tests below.

**Review by nimasmi — APPROVED** [review]
Thanks Victor, this looks good to me.

**Pull request merged**